Zend/zend_API.c

```diff
@@ -120,7 +120,7 @@ ZEND_API int _zend_get_parameters_array_ex(int param_count, zval *argument_array
 	while (param_count-->0) {
 		ZVAL_COPY_VALUE(argument_array, param_ptr);
 		argument_array++;
-		param_ptr++;
+		param_ptr--;
```

Comment:
zend_get_parameters_array() is going to be logical inconsistent with the rest or parameter parsing API.
It uses direct arguments order, while all others uses reverse order.
I think this may be solved by introducing simple arg_array API.
This should also solve mess with negative indexing and catch all affected places at compile time.
I think such arg_array API may be designed and applied before this patch.